src/objects/core/utils.py

```diff
@@ -1,37 +1,48 @@
+from django.conf import settings
 from django.core.exceptions import ValidationError
 
 import jsonschema
 import requests
 from zgw_consumers.client import build_client
 
+from objects.utils.cache import cache
+
 
 def check_objecttype(object_type, version, data):
-    client = build_client(object_type.service)
-    objecttype_version_url = f"{object_type.url}/versions/{version}"
+    @cache(
+        f"objecttypen-{object_type.uuid}:versions-{version}",
+        timeout=settings.OBJECTTYPE_VERSION_CACHE_TIMEOUT,
+    )
+    def get_objecttype_version_response():
```

Comment:
The implementation works, but I think it would be nice if we have a similar implementation as in Open Forms:

https://github.com/open-formulieren/open-forms/blob/7f5390b52572da99dbcf5c2430244e260ec098a4/src/openforms/contrib/objects_api/clients/objecttypes.py#L63

Basically an `ObjecttypesClient` class which implements `get_objecttype_version` (uncached) and another method that calls `get_objecttype_version` but is cached. If we keep the structure the same, we might be able to create a library at some point for all these common clients